airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/config/DataSourceFactory.kt

```diff
@@ -31,3 +26,37 @@ class DataSourceFactory {
         return dataSource
     }
 }
+
+fun MSSQLConfiguration.toSQLServerDataSource(): SQLServerDataSource {
+    val connectionStringArray =
```

Comment:
nit:  you could do something like this in Kotlin:

```kotlin
val connectionStringBuilder = StringBuilder().apply {
     append("jdbc:sqlserver://$host:$port;databaseName=$dtabase")
     when (sslMethod) {
         ....
         append(...)
      }
}.toString()
```

Reply:
I'll look into this as part of the next PR when I work on the datasource a bit more.